src/dparse/parser.d

```diff
@@ -4966,9 +5031,11 @@ class Parser
                 mixin(parseNodeQ!(`node.conditionalStatement`, `ConditionalStatement`));
             else if (peekIs(tok!"assert"))
                 mixin(parseNodeQ!(`node.staticAssertStatement`, `StaticAssertStatement`));
+            else if (peekIs(tok!"foreach") || peekIs(tok!"foreach_reverse"))
+                mixin(parseNodeQ!(`node.staticForeachStatement`, `StaticForeachStatement`));
             else
             {
-                error("'if' or 'assert' expected.");
+                error("'if' or 'assert' or 'foreach' or 'foreach_reverse' expected.");
```

Comment:
```error("'if', 'assert', 'foreach', or 'foreach_reverse' expected.");```

Reply:
> I'd like to see a test case that checks that the new alias and enum attributes are parsed for the foreach statement.

Thank you @Hackerpilot for the review. I didn't know there were new type attributes.